### DISCUSSION TITLE
I want to prevent the user from pressing a button to escape if the validation fails in the dialog box.

### DISCUSSION THREAD

**Discussion by dimohy** [microsoft/fluentui-blazor#699] [Q&A]
Hello

I want to validate through the edit form in the dialog, but the way I'm doing it now, when I click the dialog button, it takes me out of the dialog.

I don't want them to leave if they don't pass the validation when they hit the OK button, but I don't know how to intercept the default button click in the dialog.

Is this possible?

**Comment by vnbaaij** [comment]
In version 3.1 (due to be released coming week) we changed the Dialog component to support `HeaderTemplate` and `FooterTemplate` (both of type `RenderFragment`) parameters. This should allow you to define your own buttons and actions

**Comment by vnbaaij** [comment]
For your scenario, it is probably better to not use the footer at all. You can achieve this by making sure `PrimaryAction` and `SecondaryAction` are null/empty. Then you can add buttons to your own dialog and do validation checking yourself. The dialog can be closed in you button action by calling the `CloseAsync()` method.
